src/System.IO.FileSystem/src/System/IO/FileSystem.Unix.cs

```diff
@@ -169,6 +169,7 @@ public static void MoveFile(string sourceFullPath, string destFullPath, bool ove
                     if (errorInfo.Error == Interop.Error.EXDEV) // rename fails across devices / mount points
                     {
                         CopyFile(sourceFullPath, destFullPath, overwrite);
+                        DeleteFile(sourceFullPath);
```

Comment:
This works, but I'm not sure it's the best answer.  There's already a DeleteFile call below; seems like a better option might be to just change the below:
```C#
        // Rename or CopyFile complete
        return;
    }

    LinkOrCopyFile(sourceFullPath, destFullPath);
    DeleteFile(sourceFullPath);
}
```
to instead be:
```C#
    }
    else
    {
        LinkOrCopyFile(sourceFullPath, destFullPath);
    }

    DeleteFile(sourceFullPath);
}
```
They accomplish the same thing, but without the duplicate DeleteFile call.

Reply:
Actually, nevermind.  What you have is good.